### ISSUE TITLE
Mac OS 10.10.5 crash, help me

### ISSUE THREAD

**Issue by caimaoy** [retext-project/retext#273]
python 3.6.1, use pyenv, I add this codes in the file `window.py`

```
		if sys.platform.startswith('darwin'):
			# https://github.com/retext-project/retext/issues/198
			searchPaths = QIcon.themeSearchPaths()
			searchPaths.append('/opt/local/share/icons')
			searchPaths.append('/usr/local/share/icons')
			searchPaths.append('/Users/caimaoy/.pyenv/versions/3.6.1/share/icons')  # caimaoy  DIR
			print(searchPaths)

```

And these codes:

```
		print('b')
		menuFile.addAction(self.actionReload)
		print('a')

```

Crash, I found that `b` was printed, but `a` wasn't.

```
Using configuration file: /Users/caimaoy/.config/ReText project/ReText.ini
[':/icons', '/opt/local/share/icons', '/usr/local/share/icons', '/Users/caimaoy/.pyenv/versions/3.6.1/share/icons']
b
2017-04-03 10:14:36.567 python3.6[18327:5115186] *** Assertion failure in -[NSBitmapImageRep initWithCGImage:], /SourceCache/AppKit/AppKit-1348.17/AppKit.subproj/NSBitmapImageRep.m:1288
2017-04-03 10:14:36.583 python3.6[18327:5115186] An uncaught exception was raised
```

All stack are below:
```
2017-04-03 10:10:03.159 python3.6[15758:5106787] *** Assertion failure in -[NSBitmapImageRep initWithCGImage:], /SourceCache/AppKit/AppKit-1348.17/AppKit.subproj/NSBitmapImageRep.m:1288
2017-04-03 10:10:03.170 python3.6[15758:5106787] An uncaught exception was raised
2017-04-03 10:10:03.170 python3.6[15758:5106787] Invalid parameter not satisfying: cgImage != NULL
2017-04-03 10:10:03.171 python3.6[15758:5106787] (
	0   CoreFoundation                      0x00007fff8721803c __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x00007fff85e7f76e objc_exception_throw + 43
	2   CoreFoundation                      0x00007fff87217e1a +[NSException raise:format:arguments:] + 106
	3   Foundation                          0x00007fff8acf599b -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 195
	4   AppKit                              0x00007fff889dc3d3 -[NSBitmapImageRep initWithCGImage:] + 135
	5   QtGui                               0x00000001027ceabd _Z21qt_mac_create_nsimageRK5QIconi + 525
	6   libqcocoa.dylib                     0x000000010b55cfda qt_plugin_instance + 154170
	7   libqcocoa.dylib                     0x000000010b55a197 qt_plugin_instance + 142327
	8   QtWidgets                           0x00000001052ca58d _ZN5QMenu11actionEventEP12QActionEvent + 1213
	9   QtWidgets                           0x0000000105186ef8 _ZN7QWidget5eventEP6QEvent + 1240
	10  QtWidgets                           0x00000001052c8654 _ZN5QMenu5eventEP6QEvent + 1140
	11  QtWidgets                           0x000000010514658d _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent + 269
	12  QtWidgets                           0x0000000105148ed2 _ZN12QApplication6notifyEP7QObjectP6QEvent + 5906
	13  QtWidgets.so                        0x0000000104d182b7 _ZN15sipQApplication6notifyEP7QObjectP6QEvent + 231
	14  QtCore                              0x0000000101dd3544 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent + 164
	15  QtWidgets                           0x000000010517ac61 _ZN7QWidget12insertActionEP7QActionS1_ + 353
	16  QtWidgets.so                        0x0000000104d80d08 _ZL20meth_QMenu_addActionP7_objectS0_S0_ + 136
	17  python3.6                           0x0000000101246a74 _PyCFunction_FastCallDict + 500
	18  python3.6                           0x00000001012c71f8 call_function + 440
	19  python3.6                           0x00000001012c3bd8 _PyEval_EvalFrameDefault + 24376
	20  python3.6                           0x00000001012c7c9c _PyEval_EvalCodeWithName + 2508
	21  python3.6                           0x00000001012c8852 _PyFunction_FastCallDict + 754
	22  python3.6                           0x00000001011fe6f7 _PyObject_FastCallDict + 247
	23  python3.6                           0x00000001011fe815 _PyObject_Call_Prepend + 149
	24  python3.6                           0x00000001011fe531 PyObject_Call + 97
	25  python3.6                           0x000000010125eabe slot_tp_init + 158
	26  python3.6                           0x000000010125ae99 type_call + 313
	27  python3.6                           0x00000001011fe6c5 _PyObject_FastCallDict + 197
	28  python3.6                           0x00000001012c7102 call_function + 194
	29  python3.6                           0x00000001012c3bd8 _PyEval_EvalFrameDefault + 24376
	30  python3.6                           0x00000001012c7c9c _PyEval_EvalCodeWithName + 2508
	31  python3.6                           0x00000001012c8541 fast_function + 545
	32  python3.6                           0x00000001012c71d2 call_function + 402
	33  python3.6                           0x00000001012c3bd8 _PyEval_EvalFrameDefault + 24376
	34  python3.6                           0x00000001012c7c9c _PyEval_EvalCodeWithName + 2508
	35  python3.6                           0x00000001012bdbe4 PyEval_EvalCode + 100
	36  python3.6                           0x00000001012f1d91 PyRun_FileExFlags + 209
	37  python3.6                           0x00000001012f13c5 PyRun_SimpleFileExFlags + 469
	38  python3.6                           0x000000010130a2a2 Py_Main + 3522
	39  python3.6                           0x00000001011f37cc main + 236
	40  libdyld.dylib                       0x00007fff861575c9 start + 1
	41  ???                                 0x0000000000000002 0x0 + 2
)
2017-04-03 10:10:03.171 python3.6[15758:5106787] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Invalid parameter not satisfying: cgImage != NULL'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff8721803c __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x00007fff85e7f76e objc_exception_throw + 43
	2   CoreFoundation                      0x00007fff87217e1a +[NSException raise:format:arguments:] + 106
	3   Foundation                          0x00007fff8acf599b -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 195
	4   AppKit                              0x00007fff889dc3d3 -[NSBitmapImageRep initWithCGImage:] + 135
	5   QtGui                               0x00000001027ceabd _Z21qt_mac_create_nsimageRK5QIconi + 525
	6   libqcocoa.dylib                     0x000000010b55cfda qt_plugin_instance + 154170
	7   libqcocoa.dylib                     0x000000010b55a197 qt_plugin_instance + 142327
	8   QtWidgets                           0x00000001052ca58d _ZN5QMenu11actionEventEP12QActionEvent + 1213
	9   QtWidgets                           0x0000000105186ef8 _ZN7QWidget5eventEP6QEvent + 1240
	10  QtWidgets                           0x00000001052c8654 _ZN5QMenu5eventEP6QEvent + 1140
	11  QtWidgets                           0x000000010514658d _ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent + 269
	12  QtWidgets                           0x0000000105148ed2 _ZN12QApplication6notifyEP7QObjectP6QEvent + 5906
	13  QtWidgets.so                        0x0000000104d182b7 _ZN15sipQApplication6notifyEP7QObjectP6QEvent + 231
	14  QtCore                              0x0000000101dd3544 _ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent + 164
	15  QtWidgets                           0x000000010517ac61 _ZN7QWidget12insertActionEP7QActionS1_ + 353
	16  QtWidgets.so                        0x0000000104d80d08 _ZL20meth_QMenu_addActionP7_objectS0_S0_ + 136
	17  python3.6                           0x0000000101246a74 _PyCFunction_FastCallDict + 500
	18  python3.6                           0x00000001012c71f8 call_function + 440
	19  python3.6                           0x00000001012c3bd8 _PyEval_EvalFrameDefault + 24376
	20  python3.6                           0x00000001012c7c9c _PyEval_EvalCodeWithName + 2508
	21  python3.6                           0x00000001012c8852 _PyFunction_FastCallDict + 754
	22  python3.6                           0x00000001011fe6f7 _PyObject_FastCallDict + 247
	23  python3.6                           0x00000001011fe815 _PyObject_Call_Prepend + 149
	24  python3.6                           0x00000001011fe531 PyObject_Call + 97
	25  python3.6                           0x000000010125eabe slot_tp_init + 158
	26  python3.6                           0x000000010125ae99 type_call + 313
	27  python3.6                           0x00000001011fe6c5 _PyObject_FastCallDict + 197
	28  python3.6                           0x00000001012c7102 call_function + 194
	29  python3.6                           0x00000001012c3bd8 _PyEval_EvalFrameDefault + 24376
	30  python3.6                           0x00000001012c7c9c _PyEval_EvalCodeWithName + 2508
	31  python3.6                           0x00000001012c8541 fast_function + 545
	32  python3.6                           0x00000001012c71d2 call_function + 402
	33  python3.6                           0x00000001012c3bd8 _PyEval_EvalFrameDefault + 24376
	34  python3.6                           0x00000001012c7c9c _PyEval_EvalCodeWithName + 2508
	35  python3.6                           0x00000001012bdbe4 PyEval_EvalCode + 100
	36  python3.6                           0x00000001012f1d91 PyRun_FileExFlags + 209
	37  python3.6                           0x00000001012f13c5 PyRun_SimpleFileExFlags + 469
	38  python3.6                           0x000000010130a2a2 Py_Main + 3522
	39  python3.6                           0x00000001011f37cc main + 236
	40  libdyld.dylib                       0x00007fff861575c9 start + 1
	41  ???                                 0x0000000000000002 0x0 + 2
)
libc++abi.dylib: terminating with uncaught exception of type NSException

```

**Comment by mitya57** [comment]
Hi @caimaoy!

First, no need to use debug prints. You could run your Python interpreter with `-X faulthandler` option, and it would print the Python traceback of where the crash happened.

Then, can you please figure out where the `view-refresh` icon comes from, and whether that’s a valid file? Maybe you can attach it here?

**Comment by markusvoelter** [comment]
I get that same exception directly after startup of Retext.

**Comment by antiphasis** [comment]
Following the advice from mitya57 I've started retext with `python3 -X faulthandler retext.py`. 

before that i satisfied all given dependencies, downloaded the repo with `git clone`
```
libc++abi.dylib: terminating with uncaught exception of type NSException
Fatal Python error: Aborted

Current thread 0x00007fffa7b143c0 (most recent call first):
  File "/Users/abczyx/devel/python3/retext/ReText/window.py", line 255 in __init__
  File "retext.py", line 79 in main
  File "retext.py", line 111 in <module>
Abort trap: 6
```

**Comment by tristanfisher** [comment]
For me, this crashes reliably on `ReText/window.py(263)__init__()` (`menuFile.addAction(self.actionReload)` a QAction object) and sounds suspiciously like https://bugreports.qt.io/browse/QTBUG-58344 .

**Comment by mitya57** [comment]
@tristanfisher Thanks a lot for the Qt bug link! Looks like it is already fixed in Qt 5.9. Unfortunately PyQt decided to skip 5.9.0 release and only release a wheel for 5.9.1, so we may need to wait about a month until new PyQt is released.

**Comment by tristanfisher** [comment]
@mitya57 No problem at all and sorry for not adding re-pro steps/proof. Thank you for actively maintaining the project.

**Comment by mitya57** [comment]
The crash should be fixed with PyQt 5.9, so closing this bug. If it still happens with Qt/PyQt 5.9 then please let me know.
